### PR TITLE
[inductor] replace ops.div with ops.truediv

### PR DESCRIPTION
**Pull request by jgong5**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112243
* #112234

cc @voznesenskym @penguinwu @EikanWang @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Review by lezcano — COMMENTED** [review]
The idea behind `div` was to be like `operator/` in C++ and `truediv` to be like `/` in Python 3, that is, with respect to type promotion.

How does this affect that point? Shouldn't we remove `truediv` rather than `div` here?

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -1546,7 +1546,7 @@ def _mask_value(value, default):
                 sum_ = ops.reduction(dtype, dtype, "sum", value)
                 self.inside_reduction = False
                 rnumel = ops.index_expr(self.numels[-1], dtype)
-                mean = ops.div(sum_, rnumel)
+                mean = ops.truediv(sum_, rnumel)
```
Comment (peterbell10) [inline torch/_inductor/codegen/triton.py:1549]: I'm confused where truediv is defined for triton? The tests are clearly passing but I can't find it anywhere.

Comment (peterbell10) [reply torch/_inductor/codegen/triton.py:1549]: Oh I think it fallls all the way back to `MockHandler` in which case it definitely does `a / b` which in triton is C division not truediv.

Comment (peterbell10) [reply torch/_inductor/codegen/triton.py:1549]: Actually I take it back, it is true div in triton. It's floor div that triton handles incorrectly.

**Review by lezcano — APPROVED** [review]
If the triton bit is good, I think this should be good (should be pretty much a no-op, as all those functions don't accept integer tensors anyway, and if they did, I reckon that `truediv` is what we'd want to use there.

**Pull request closed without merging**